force-app/main/default/classes/OpportunityUtils.cls

```diff
@@ -0,0 +1,60 @@
+/*
+ * Additional methods needed for the OpportunityTriggerHandler class
+ * 
+ * @author Ilana Shotkin
+*/
+
+public with sharing class OpportunityUtils {
+    /* 
+     * Retrieve a List of all Contacts for Opportunities' Accounts with a specific Title
+     * 
+     * @param oppList The list of all Opportunities that need to have a Contact located
+     * @param title The Title that the Contacts must have
+     * @return A list of all Contacts fitting the criteria
+     */
+    public static List<Contact> getContactsByTitle(List<Opportunity> oppList, String title) {
+        // Get contacts related to Opportunities with a certain title
+        List <Contact> contactList = 
+            [SELECT Id, FirstName, AccountId 
+            FROM Contact 
+            WHERE AccountId IN (SELECT AccountID FROM Opportunity WHERE Id IN :oppList) AND Title = :title 
```

Comment:
Even though it works, it is not recommended to pass a list of objects directly into a query. Better to extract the ids into a Set first. I am not sure the exact technological explanation, but I know this method is called out as less efficient.